src/components/custom/array-field/array-field-element.tsx

```diff
@@ -0,0 +1,81 @@
+import isEqual from "lodash/isEqual";
+import { ForwardedRef, forwardRef, useEffect, useImperativeHandle, useRef, useState } from "react";
+import { useFormState } from "react-hook-form";
+import { usePrevious } from "../../../utils/hooks";
+import { FrontendEngine } from "../../frontend-engine";
+import { IFrontendEngineRef, TFrontendEngineFieldSchema, TFrontendEngineValues } from "../../types";
+
+interface ArrayFieldElementProps {
+	onChange: (data: TFrontendEngineValues, isFormValid: boolean) => void;
+	schema: Record<string, TFrontendEngineFieldSchema>;
+	formValues?: TFrontendEngineValues;
+}
+
+const Component = ({ onChange, formValues, schema }: ArrayFieldElementProps, ref: ForwardedRef<IFrontendEngineRef>) => {
+	// =============================================================================
+	// CONST, STATE, REFS
+	// =============================================================================
+	const formRef = useRef<IFrontendEngineRef>(null);
+	const [sectionValues, setSectionValues] = useState<TFrontendEngineValues>({});
+	const [defaultValues, setDefaultValues] = useState<TFrontendEngineValues>();
+	const { isSubmitting, isDirty } = useFormState();
+	const prevIsDirty = usePrevious(isDirty);
+
+	useImperativeHandle<IFrontendEngineRef, IFrontendEngineRef>(ref, () => formRef.current);
+
+	// =============================================================================
+	// EFFECTS
+	// =============================================================================
+	useEffect(() => {
+		if (!isDirty && prevIsDirty !== undefined && prevIsDirty !== isDirty) {
+			formRef.current?.reset();
+		}
+	}, [isDirty]);
+
+	useEffect(() => {
+		if (!isDirty) {
+			setDefaultValues(formValues);
+		}
+
+		if (isEqual(formValues, sectionValues)) {
+			return;
+		}
+
+		if (formValues) {
+			Object.entries(formValues).forEach(([key, value]) => {
+				formRef.current?.setValue(key, value);
+			});
+		}
```

Comment:
found an issue with empty `formValues`
1. add a new section
2. key in value in 1st section and leave 2nd section empty
3. delete 1st section
4. the values from 1st section got transferred to the 2nd because it is not able to clear it

we either need to handle empty `formValues` or force a remount

Reply:
I wasn't able to clear the form using react-hook-form `reset()` due to the use of default values. I also didn't want to replicate the code in `reset-button` to manually iterate through the keys

instead, I added a unique key identifier for each section to prevent React from unmounting. this helps to preserve the state in each section and prevents unnecessary updates of `formValues` prop